components/ide/jetbrains/backend-plugin/src/main/kotlin/io/gitpod/jetbrains/remote/GitpodManager.kt

```diff
@@ -365,4 +365,31 @@ class GitpodManager : Disposable {
             }
         }
     }
+
+    var resourceStatus: Status.ResourcesStatusResponse? = null
+
+    private val metricsJob = GlobalScope.launch {
+        if (application.isHeadlessEnvironment) {
+            return@launch
+        }
+        val status = StatusServiceGrpc.newFutureStub(supervisorChannel)
+        while (isActive) {
+            try {
+                val f = status.resourcesStatus(Status.ResourcesStatuRequest.getDefaultInstance())
+                resourceStatus = f.asDeferred().await()
+            } catch (t: Throwable) {
+                if (t is CancellationException) {
+                    throw t
+                }
+                thisLogger().error("gitpod: failed to retrieve resource status: ", t)
+            }
+            delay(1000L)
```

Comment:
I've noticed the rate-limiting sometimes preventing to receive the fresh metrics. In the meantime that we work on supervisor for extending the rate-limiting capabilities, shall we reduce the refresh to `5s`?

Reply:
It would be implement exponential backoff, you can check how gitpod-monitor does it: https://github.com/akosyakov/gitpod-monitor/blob/15bc5a39d8ab4f71a18ce7e0a38a70c8e9e0e526/src/extension.ts#L65-L77

Reply:
@andreafalzetti Could you create an issue please that we improve stability of this supervisor endpoint? I think we should move polling each second in supervisor and returned cached value to all clients. It will greatly simplify clients.

Reply:
Issue created, I would appreciate a double check of the description to make sure it makes sense 😄 [[1]](https://github.com/gitpod-io/gitpod/issues/12083)